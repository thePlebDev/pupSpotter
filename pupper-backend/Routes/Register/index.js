const express = require('express');
const registerUserService = require('../../Services/registerUserService')

const registerRouter = express.Router()

registerRouter.post('/', (req,res,next)=>{
  const {username,password,bio,email} = req.body
  registerUserService.registerUser(username,password,bio,email,next)
    .then(data=>{
      // console.log(data)
      res.json(data)
    })
    .catch(error=>{next(error)})
})


module.exports = registerRouter
