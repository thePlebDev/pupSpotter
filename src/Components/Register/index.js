import React,{useState,useEffect,useRef} from 'react';
import {makeStyles} from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import Button from '@material-ui/core/Button';
import styled from 'styled-components';

import useRegister from '../../Hooks/UseRegister'
import {registerValidation} from '../../utils/Validation';

import {Form,TextFieldContainer,Title,FormContentContainer} from '../../Assets/FormStylings';





const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(1),
      'font-size':'1.2em',
    },
    marginBottom:'20px',
  },
  alerts: {
    width: '100%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
  },
  buttons: {
    width:'70%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
    padding:'10px',
    border:'1px solid #05386B',
    marginBottom:'20px',
    marginLeft:'15%'

  },

}));
const Image = styled.img`
  display:none;
  @media screen and (min-width: 992px) {
  display:block;
}
`



const Register =(props)=>{
  const {state,bio,email,password,errors,handleChange,handleSubmit,status,setStatus} = useRegister(registerValidation)
  const classes = useStyles()

  return(
    <div style={{height:'100%',display:'flex',alignItems:'center'}}>
      <Image src={require("../../Assets/images/ladyDog.png")} style={{width:'50%'}}  alt="dog in nature" />
      <Form onSubmit={(e)=>{handleSubmit(e)}} >
      <FormContentContainer>
          <Title>Register</Title>

          <TextFieldContainer>
            <TextField className={classes.root} id="standard-basic" label="Username" style={{width:'70%',marginLeft:'15%',padding:'5px'}}
            error={errors.username} value={state} onChange={(e)=>handleChange(e)} name='username'/>
          </TextFieldContainer>

          <TextFieldContainer>
            <TextField className={classes.root} id="standard-basic" label="Email" style={{width:'70%',marginLeft:'15%'}}
            name="email" error={errors.email}  value={email} onChange={(e)=>handleChange(e)}/>
          </TextFieldContainer>

          <TextFieldContainer>
            <TextField className={classes.root} id="standard-basic" label="Bio" style={{width:'70%',marginLeft:'15%'}}
            name="bio" value={bio}onChange={(e)=>handleChange(e)}/>
          </TextFieldContainer
          >
          <TextFieldContainer>
            <TextField className={classes.root} id="standard-basic" label="Password" style={{width:'70%',marginLeft:'15%'}}
            name="password" error={errors.password}  type='password' value={password}onChange={(e)=>handleChange(e)}/>
          </TextFieldContainer>
          <Button className={classes.buttons}  type="Submit" disableRipple={true}>Submit</Button>
          
        </FormContentContainer>
      </Form>
    </div>
  )
}

export default Register
