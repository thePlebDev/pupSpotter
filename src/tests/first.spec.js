import React from 'react';
import Register from '../Components/Register';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';


describe('testing',()=>{
  test('testing snapshot of SideBar',()=>{
    const tree = renderer
      .create(<Register></Register>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
