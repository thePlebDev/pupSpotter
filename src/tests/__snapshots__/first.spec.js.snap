// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing testing snapshot of SideBar 1`] = `
<div
  className="sc-Axmtr gyjOmI"
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="sc-fzozJi hHVLEE"
    >
      <span>
        Username Taken!
      </span>
    </div>
  </div>
  <form
    className="sc-AxhCb fjWdks"
    onSubmit={[Function]}
  >
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": "30px",
            "position": "relative",
          }
        }
      >
        <input
          className="sc-AxmLO jkpzj"
          name="username"
          onChange={[Function]}
          type="text"
        />
        <label
          className="sc-AxirZ iRSiUU"
          for="username"
        >
          <span
            className="sc-AxjAm iJGPez"
          >
            Username
          </span>
        </label>
      </div>
    </div>
    <div
      style={
        Object {
          "marginBottom": "30px",
          "position": "relative",
        }
      }
    >
      <input
        className="sc-AxmLO jkpzj"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        className="sc-AxirZ iRSiUU"
        for="email"
      >
        <span
          className="sc-AxjAm iJGPez"
        >
          Email
        </span>
      </label>
    </div>
    <div
      style={
        Object {
          "marginBottom": "30px",
          "position": "relative",
        }
      }
    >
      <input
        className="sc-AxmLO jkpzj"
        name="bio"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        className="sc-AxirZ iRSiUU"
        for="bio"
      >
        <span
          className="sc-AxjAm iJGPez"
        >
          Bio
        </span>
      </label>
    </div>
    <div
      style={
        Object {
          "marginBottom": "30px",
          "position": "relative",
        }
      }
    >
      <input
        className="sc-AxmLO jkpzj"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <label
        className="sc-AxirZ iRSiUU"
        for="password"
      >
        <span
          className="sc-AxjAm iJGPez"
        >
          Password
        </span>
      </label>
    </div>
    <button
      className="sc-AxgMl eeblGN"
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
